### ISSUE TITLE
Implement IQueryable Extensions

### ISSUE THREAD

**Issue by brandongregoryscott** [AndcultureCode/AndcultureCode.CSharp.Extensions#15]
After running SQL Profiler, we noticed that using the `IEnumerable.IsNullOrEmpty()` extension method in this repo will run a full `select` on a table and _then_ evaluate whether or not there are elements in memory, vs. the `IQueryable.Any()` which modifies the `select` like so:

```
exec sp_executesql N'SELECT CASE
    WHEN EXISTS (
        SELECT 1
        FROM [UserRoles] AS [e]
        WHERE ([e].[DeletedOn] IS NULL AND ([e].[UserId] = @__userId_0)) AND ([e].[EulaAccepted] = 0))
    THEN CAST(1 AS BIT) ELSE CAST(0 AS BIT)
END',N'@__userId_0 bigint',@__userId_0=4
```

At the very least, we should implement an `IsEmpty()`, and `IsNullOrEmpty()` for `IQueryable` (which call the `IQueryable.Any()` method), to have parity with `IEnumerable`. I imagine there are many places where we're checking `IsNullOrEmpty()` on an `IQueryable` `ResultObject` and it is doing that check in memory instead of in the query.

I'd be open to other suggestions for extensions as well.

**Comment by wintondeshong** [comment]
@brandongregoryscott how does this play out from an implementation perspective? I'm assuming being IQueryable<T> inherits from IEnumerable<T> and they are both in the same namespace, it picks the closest/exact (in this case) match of typed extensions. IQueryable<T>.XYZ vs. IEnumerable<T>.XYZ. Just verifying.

**Comment by wintondeshong** [comment]
Published with v0.2.0
